### PR TITLE
pascals-triangle: split into instructions and introduction

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: exercises/pascals-triangle/instructions.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Instructions
+
+Your task is to output the first N rows of Pascal's Triangle.
```
Comment (ErikSchierboom) [inline exercises/pascals-triangle/instructions.md:3]: Do we need more here? Some examples?

File: exercises/pascals-triangle/introduction.md
Comment (ErikSchierboom) [inline exercises/pascals-triangle/introduction.md:1]: I couldn't come up with a story here. I'm open to suggestions :)

Comment (jiegillet) [reply exercises/pascals-triangle/introduction.md:1]: uh, I didn't even know you could comment on file, TIL.

Pascal's Triangle is a classic: here are some stuff you can do with it, maybe inspiration will strike:
- You can draw a Sierpinski triangle by coloring the even/odd numbers
- the sum of each row is a power of 2
- each element is a combination, entry k in row n tells you how many ways you can pick k elements from n
- squishing all the numbers in a row gives you a power of 11 (1331 = 11^3, 14641 = 11^4)

It's all a bit nerdy, but that can be a good thing :)

Comment (ErikSchierboom) [reply exercises/pascals-triangle/introduction.md:1]: I think the third one is most suitable for a story. Thanks

File: exercises/pascals-triangle/instructions.md (hunk 1 above)
Comment (BNAndras) [inline exercises/pascals-triangle/instructions.md:3]:
```suggestion
Your task is to output the first N rows of Pascal's triangle.
```
Minor capitalization correction in line with the introduction and how Wikipedia cases it.

Comment (BNAndras) [reply exercises/pascals-triangle/instructions.md:3]: An example explicitly showing the addition process like in https://en.wikipedia.org/wiki/Pascal%27s_triangle#/media/File:PascalTriangleAnimated2.gif might be handy. I think that's where students might get stuck. Once you've got over that, I don't think additional examples provide a lot of extra insight. The modeled triangle doesn't change, just the number of rows exported.

File: exercises/pascals-triangle/introduction.md
Comment (IsaacG) [reply exercises/pascals-triangle/introduction.md:1]: I had no idea there were so many things you could do with it!

Comment (jiegillet) [reply exercises/pascals-triangle/introduction.md:1]: Oh, there's more, there are so many patterns in these numbers. Fibonacci numbers are also hiding in there.

Comment (MatthijsBlom) [reply exercises/pascals-triangle/introduction.md:1]: Much more!

- An especially accessible list of patterns: [Pascal's Triangle @ MathIsFun](https://www.mathsisfun.com/pascals-triangle.html). Possibly interesting exercise: spot the duplicates. Most of the patterns listed here are really the very same pattern, just phrased in different ways.

- Skim the images in [the Wikipedia article](https://en.wikipedia.org/wiki/Pascal's_triangle) for hints at more patterns, elaborated on in the text.

**Comment by ErikSchierboom** [comment]
@BNAndras I've added a bit more text, but so far struggled to visually and textually explain the logic. I'm open to suggestions.

**Comment by ErikSchierboom** [comment]
I've added a story and updated the instructions.

File: exercises/pascals-triangle/introduction.md [hunk 2]
```diff
@@ -1,25 +1,24 @@
...
+[Pascal's triangle][wikipedia].
 
-The values in rows are computed by adding the numbers directly to the right and left of the current position in the previous row.
-If the previous row does not have a value to the left or right of the current position, treat that value as zero.
+Over the next hour, your teacher reveals some amazing things hidden in this triangle:
 
-The one exception is the first (topmost) row (as it does not have a previous row) and which contains a single one.
+- It can be used to compute how many ways you can pick K elements from N values.
+- It contains the Fibonacci sequence.
+- If you color odd and even number differently, you get a beautiful pattern called the [Sierpiński triangle][wikipedia-sierpinski-triangle]
```
Comment (BethanyG) [inline exercises/pascals-triangle/introduction.md:15]:
```suggestion
- If you color odd and even numbers differently, you get a beautiful pattern called the [Sierpiński triangle][wikipedia-sierpinski-triangle]
```

File: exercises/pascals-triangle/introduction.md [hunk 3]
```diff
@@ -1,25 +1,24 @@
...
 
-## Example
+The teacher implores you and your classmates to lookup other other uses, and assures you that there are lots more!
+At that moment, the school bell rings.
+You realize that for the past hour, you were completely absorbed in learning about Pascal's triangle. Math is definitely not as boring as you thought it was!
```
Comment (BethanyG) [inline exercises/pascals-triangle/introduction.md:19]:
```suggestion
You realize that for the past hour, you were completely absorbed in learning about Pascal's triangle.
Math is definitely not as boring as you thought it was!
```

**Review by BethanyG — APPROVED** [review]

File: exercises/pascals-triangle/introduction.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+Not long after you sit down, your teacher enters the room and explains that this triangle is the famous
+[Pascal's triangle][wikipedia].
+
+Over the next hour, your teacher reveals some amazing things hidden in this triangle:
+
+- It can be used to compute how many ways you can pick K elements from N values.
+- It contains the Fibonacci sequence.
+- If you color odd and even numbers differently, you get a beautiful pattern called the [Sierpiński triangle][wikipedia-sierpinski-triangle]
+
+The teacher implores you and your classmates to lookup other other uses, and assures you that there are lots more!
```
Comment (ErikSchierboom) [inline exercises/pascals-triangle/introduction.md:17]:
```suggestion
The teacher implores you and your classmates to lookup other uses, and assures you that there are lots more!
```

File: exercises/pascals-triangle/introduction.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+Over the next hour, your teacher reveals some amazing things hidden in this triangle:
+
+- It can be used to compute how many ways you can pick K elements from N values.
+- It contains the Fibonacci sequence.
+- If you color odd and even numbers differently, you get a beautiful pattern called the [Sierpiński triangle][wikipedia-sierpinski-triangle]
+
+The teacher implores you and your classmates to lookup other uses, and assures you that there are lots more!
+At that moment, the school bell rings.
+You realize that for the past hour, you were completely absorbed in learning about Pascal's triangle.
+Math is definitely not as boring as you thought it was!
```
Comment (jiegillet) [inline exercises/pascals-triangle/introduction.md:20]: As someone who doesn't think math is boring, I resent someone telling me that I thought that it was. 
How about something a bit more positive to build on the previous sentence?
 ```suggestion
Math can be really interesting after all!
```

Comment (ErikSchierboom) [reply exercises/pascals-triangle/introduction.md:20]: For the record: I also don't think math is boring (at all), I just couldn't come up with a better phrasing.

Comment (ErikSchierboom) [reply exercises/pascals-triangle/introduction.md]: I've just removed the section.

File: exercises/pascals-triangle/introduction.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+Weird!
+
+Not long after you sit down, your teacher enters the room and explains that this triangle is the famous
+[Pascal's triangle][wikipedia].
+
+Over the next hour, your teacher reveals some amazing things hidden in this triangle:
+
+- It can be used to compute how many ways you can pick K elements from N values.
+- It contains the Fibonacci sequence.
+- If you color odd and even numbers differently, you get a beautiful pattern called the [Sierpiński triangle][wikipedia-sierpinski-triangle]
```
Comment (jiegillet) [inline exercises/pascals-triangle/introduction.md:15]:
```suggestion
- If you color odd and even numbers differently, you get a beautiful pattern called the [Sierpiński triangle][wikipedia-sierpinski-triangle].
```

**Review by jiegillet — APPROVED** [review]

**Review by ConeDragon — CHANGES_REQUESTED** [review]
After discussing about the "math is boring" part, I found this on line 3:
`With the weather being great, you're not looking forward to spending an hour in a classroom doing math.`

File: exercises/pascals-triangle/introduction.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
+# Introduction
+
+With the weather being great, you're not looking forward to spending an hour in a classroom doing math.
+Annoyed, you enter the class room, where you notice a strangely satisfying triangle shape on the blackboard.
+Whilst waiting for your math teacher to arrive, you can't help but notice some patterns in the triangle: the outer values are all ones, each subsequent row has one more value than its previous row and the triangle is symmetrical.
+Weird!
+
+Not long after you sit down, your teacher enters the room and explains that this triangle is the famous
+[Pascal's triangle][wikipedia].
```
Comment (kotp) [inline exercises/pascals-triangle/introduction.md:9]:
```suggestion
Not long after you sit down, your teacher enters the room and explains that this triangle is the famous [Pascal's triangle][wikipedia].
```

File: exercises/pascals-triangle/introduction.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
+# Introduction
+
+With the weather being great, you're not looking forward to spending an hour in a classroom doing math.
+Annoyed, you enter the class room, where you notice a strangely satisfying triangle shape on the blackboard.
+Whilst waiting for your math teacher to arrive, you can't help but notice some patterns in the triangle: the outer values are all ones, each subsequent row has one more value than its previous row and the triangle is symmetrical.
```
Comment (kotp) [inline exercises/pascals-triangle/introduction.md:5]:
```suggestion
Whilst waiting for your math teacher to arrive, you can't help but notice some patterns in the triangle:
the outer values are all ones, each subsequent row has one more value than its previous row and the triangle is symmetrical.
```

**Review by kotp — APPROVED** [review]
Style consistency for "single sentence per line" regarding a link given, the change will match the link on line 15 (prior to change being made).

Broke up the long line with the colon being a separator.  While a single sentence, this is a reasonable exception to the strict "one sentence per line" and still supports the reason why we have that style adopted, I think.

**Comment by jiegillet** [comment]
> After discussing about the "math is boring" part, I found this on line 3: `With the weather being great, you're not looking forward to spending an hour in a classroom doing math.`

I personally thought that this was fine, because one can enjoy math but still enjoy great weather more.
Maybe removing "doing math" could emphasize that this is really about the weather.

File: exercises/pascals-triangle/introduction.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
+# Introduction
+
+With the weather being great, you're not looking forward to spending an hour in a classroom doing math.
```
Comment (ErikSchierboom) [inline exercises/pascals-triangle/introduction.md:3]:
```suggestion
With the weather being great, you're not looking forward to spending an hour in a classroom.
```

**Comment by ErikSchierboom** [comment]
I've updated!

**Pull request merged**